### PR TITLE
feat: add usePortalNavigation hook

### PR DESCRIPTION
**Pull request by anxobotana**
DRAFT

Adds the `usePortalNavigation` hook responsible to get the top level navigation in sync.

CU-86948pc1y

### CODE REVIEW

File: packages/app-bridge/src/AppBridgeTheme.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ import {
     type TargetsUpdate,
     type ThemeTemplate,
 } from './types';
+import { NavigationTree } from './types/NavigationTree';
```
Comment (Kenny806) [inline packages/app-bridge/src/AppBridgeTheme.ts:56]:
```suggestion
import { type NavigationTree } from './types/NavigationTree';
```

File: packages/app-bridge/src/react/usePortalNavigation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+import { renderHook, waitFor } from '@testing-library/react';
+import { getAppBridgeThemeStub } from '../tests';
+import { usePortalNavigation } from './usePortalNavigation';
+import { NavigationTreeDummy } from '../tests/NavigationTreeDummy';
+
+const DEFAULT_NAVIGATION_TREE = NavigationTreeDummy.default();
+
+describe('usePortalNavigation', () => {
+    afterEach(() => {
+        vi.resetAllMocks();
```
Comment (Kenny806) [inline packages/app-bridge/src/react/usePortalNavigation.spec.ts:15]: Unless I'm missing something, there are no mocks that should be reset - each test case initialises its own mocks.

File: packages/app-bridge/src/react/usePortalNavigation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+        const { result } = renderHook(() => usePortalNavigation(appBridge));
+
+        expect(spy).toHaveBeenCalledOnce();
+        await waitFor(() => {
+            expect(result.current.isLoading).toBe(false);
+            expect(result.current.navigationItems).toEqual(DEFAULT_NAVIGATION_TREE);
+        });
+
+        const groupIndex = DEFAULT_NAVIGATION_TREE.findIndex((item) => item.id() === 101);
```
Comment (Kenny806) [inline packages/app-bridge/src/react/usePortalNavigation.spec.ts:110]:
```suggestion
        const documentIndex = DEFAULT_NAVIGATION_TREE.findIndex((item) => item.id() === 101);
```

File: packages/app-bridge/src/react/usePortalNavigation.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+    useEffect(() => {
+        if (options.enabled) {
+            refetch();
+        }
+    }, [options.enabled, refetch]);
+
+    useEffect(() => {
+        const handleCoverPageAction = (event: EmitterEvents['AppBridge:GuidelineCoverPage:Action']) => {
+            switch (event.action) {
```
Comment (Kenny806) [inline packages/app-bridge/src/react/usePortalNavigation.ts:35]: For all the `handle*Action` callbacks:
 - the `switch` statement is only valid for one case, the remaining possibilities are no-op. How about replacing them with `if (event.action === 'delete')` and just forgetting about the remaining two?
 - the body is almost identical for all three - could it be just one callback for cover page, document and document group?
 - could the handlers be defined outside `usePortalNavigation`, similarly to `previewItemPositions`?

File: packages/app-bridge/src/react/usePortalNavigation.spec.ts (hunk 2 above)
Comment (anxobotana) [reply packages/app-bridge/src/react/usePortalNavigation.spec.ts:15]: yup... you are totally right

File: packages/app-bridge/src/react/usePortalNavigation.ts (hunk 4 above)
Comment (anxobotana) [reply packages/app-bridge/src/react/usePortalNavigation.ts]: I did it to be explicit... but you are right, refactored :)

**Review by Kenny806 — APPROVED** [review]
LGTM

File: packages/app-bridge/src/react/usePortalNavigation.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { produce } from 'immer';
+import debounce from 'lodash-es/debounce';
+import { useCallback, useEffect, useState } from 'react';
+
+import { type AppBridgeTheme } from 'src/AppBridgeTheme';
+import { type EmitterEvents } from 'src/types/Emitter';
+import { type NavigationTreeClassTypes, type NavigationTree } from 'src/types/NavigationTree';
```
Comment (SamuelAlev) [inline packages/app-bridge/src/react/usePortalNavigation.ts:9]: Can you fix the import to be relative?

File: packages/app-bridge/src/react/usePortalNavigation.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                        }
+                        return draft;
+                    }),
+                );
+            }
+
+            debounce(refetch, 100);
+        };
+
+        const handlerDocumentGroupMoveEventPreview = (
```
Comment (mike85) [inline packages/app-bridge/src/react/usePortalNavigation.ts:74]:
```suggestion
        const handleDocumentGroupMoveEventPreview = (
```

File: packages/app-bridge/src/types/NavigationTree.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    type GuidelineDocumentLink,
+} from './Guideline';
+
+export type NavigationTree = (
+    | GuidelineCoverPage
+    | GuidelineDocumentGroup
+    | GuidelineDocument
+    | GuidelineDocumentLibrary
+    | GuidelineDocumentLink
+)[];
```
Comment (mike85) [inline packages/app-bridge/src/types/NavigationTree.ts:17]:
```suggestion
export type PortalNavigationItem = GuidelineCoverPage | GuidelineDocumentGroup | GuidelineDocument | GuidelineDocumentLibrary | GuidelineDocumentLink;
```

File: packages/app-bridge/src/types/NavigationTree.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+export type NavigationTreeClassTypes = (
+    | GuidelineCoverPage
+    | GuidelineDocumentGroup
+    | GuidelineDocument
+    | GuidelineDocumentLibrary
+    | GuidelineDocumentLink
+)['type'];
```
Comment (mike85) [inline packages/app-bridge/src/types/NavigationTree.ts:25]:
```suggestion
export type PortalNavigationItemTypes = PortalNavigationItem['type'];
```

File: .changeset/silver-papayas-jump.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
+"@frontify/app-bridge": patch
+---
+
+feat: add usePortalNavigation hook
```
Comment (ragi96) [inline .changeset/silver-papayas-jump.md:5]:
```suggestion
feat(AppBridgeTheme): add usePortalNavigation hook
```
Would scope it :)

**Review by ragi96 — APPROVED** [review]
Just a nit on the changeset, owned files lgtm

File: packages/app-bridge/src/tests/NavigationTreeDummy.ts [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+    static with(
+        id: number,
+        title = 'guideline document library',
+        parentId: Nullable<number> = null,
+    ): GuidelineDocumentLinkClass {
+        return new GuidelineDocumentLinkClass(id, title, parentId);
+    }
+}
+
+export class NavigationTreeDummy {
```
Comment (olivereisenhut) [inline packages/app-bridge/src/tests/NavigationTreeDummy.ts:219]: Perhaps the dummy could be named accordingly.

File: packages/app-bridge/src/types/Guideline.ts [hunk 11]
```diff
@@ -78,3 +78,12 @@ export interface GuidelineDocumentPageHeading {
...
 }
+
+export type PortalNavigationItem =
+    | GuidelineCoverPage
+    | GuidelineDocumentGroup
+    | GuidelineDocument
+    | GuidelineDocumentLibrary
+    | GuidelineDocumentLink;
+
+export type PortalNavigationItemTypes = PortalNavigationItem['type'];
```
Comment (olivereisenhut) [inline packages/app-bridge/src/types/Guideline.ts:89]: It seems sub-optimal to have two representations of the same type. Let's see how this develops.

**Review by olivereisenhut — APPROVED** [review]

**Pull request merged**